awx/ui/client/lib/components/layout/side-nav.directive.js

```diff
@@ -4,14 +4,24 @@ function atSideNavLink (scope, element, attrs, ctrl) {
     scope.layoutVm = ctrl;
 }
 
-function AtSideNavController () {
+function AtSideNavController ($scope) {
     let vm = this || {};
 
     vm.isExpanded = false;
 
     vm.toggleExpansion = () => {
         vm.isExpanded = !vm.isExpanded;
     }
+
+    document.body.onclick = (e) => {
+        if ($(e.target).parents(".at-Layout-side").length === 0) {
+            vm.isExpanded = false;
+        }
+    }
+
+    $scope.$on('$locationChangeStart', function(event) {
```

Comment:
I'd probably move this up to the layout directive where the rest of the getting data-from-outside-sources stuff is for now.  That way it's easy to abstract into a model style pattern in the future.